src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -159,14 +163,26 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
                             }
                         }
 
+                        JsonPropertyInfo[] cacheArray;
                         if (DetermineExtensionDataProperty(cache))
                         {
                             // Remove from cache since it is handled independently.
                             cache.Remove(DataExtensionProperty.NameAsString);
+
+                            cacheArray = new JsonPropertyInfo[cache.Count + 1];
+
+                            // Set the last element to the extension property.
+                            cacheArray[cache.Count] = DataExtensionProperty;
```

Comment:
It avoids an extra call to HandleObject() which is now inlined (so we only have one call now, instead of two).

FYI the dictionary-based cache doesn't (and can't) contain the extension property since it is used during deserialization to match property names which we don't want to do (it is not possible to deserialize directly into the extension property - only indirectly through a missing property)